orca-spring-batch/src/main/java/com/netflix/spinnaker/orca/batch/SpringBatchExecutionRunner.java

```diff
@@ -42,35 +44,48 @@
 import static java.lang.String.format;
 
 public class SpringBatchExecutionRunner extends ExecutionRunnerSupport {
-
+  private final ExecutionRepository executionRepository;
   private final JobLauncher jobLauncher;
   private final JobRegistry jobRegistry;
   private final JobBuilderFactory jobs;
   private final StepBuilderFactory steps;
   private final TaskTaskletAdapter taskTaskletAdapter;
-  private final Map<String, com.netflix.spinnaker.orca.Task> tasks;
+  private final Map<Class, com.netflix.spinnaker.orca.Task> tasks;
 
   public SpringBatchExecutionRunner(
     Collection<StageDefinitionBuilder> stageDefinitionBuilders,
+    ExecutionRepository executionRepository,
     JobLauncher jobLauncher,
     JobRegistry jobRegistry,
     JobBuilderFactory jobs,
     StepBuilderFactory steps,
     TaskTaskletAdapter taskTaskletAdapter,
-    Map<String, com.netflix.spinnaker.orca.Task> tasks) {
+    Collection<com.netflix.spinnaker.orca.Task> tasks
```

Comment:
A collection is fine as there's no unique name that a Task implementation can be referred by ... rather we'll just look it up by `Class`